pkg/scalers/aws_dynamodb_scaler.go

```diff
@@ -249,12 +252,75 @@ func json2Map(js string) (m map[string]*string, err error) {
 	return m, err
 }
 
-// json2DynamoMap converts Json to map[string]*dynamoDb.AttributeValue
-func json2DynamoMap(js string) (m map[string]*dynamodb.AttributeValue, err error) {
-	err = json.Unmarshal([]byte(js), &m)
-
+// json2DynamoMap converts Json to map[string]types.AttributeValue
+func json2DynamoMap(js string) (map[string]types.AttributeValue, error) {
+	var valueMap map[string]interface{}
+	err := json.Unmarshal([]byte(js), &valueMap)
 	if err != nil {
-		return nil, fmt.Errorf("%v: %w", ErrAwsDynamoInvalidExpressionAttributeValues, err)
+		return nil, err
 	}
-	return m, err
+	attributeValues := make(map[string]types.AttributeValue)
+
+	// Iterate through the input map and convert values to AttributeValues
+	for k, v := range valueMap {
+		av, err := attributeValueFromInterface(v)
+		if err != nil {
+			return nil, err
+		}
+		attributeValues[k] = av
+	}
+	return attributeValues, nil
+}
+
+func attributeValueFromInterface(value interface{}) (types.AttributeValue, error) {
```

Comment:
Does the SDK have something like this internally?

Reply:
@JorTurFer Found this package. But few test cases are failing when using that.

"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"

Reply:
Do they fail because the behavior has changed?